pkg/build/controller/build/build_controller.go

```diff
@@ -1528,6 +1551,129 @@ func (bc *BuildController) createBuildRegistryConfigMapSpec(build *buildv1.Build
 	return cm
 }
 
+func (bc *BuildController) controllerConfigWorker() {
+	for {
+		if quit := bc.controllerConfigWork(); quit {
+			return
+		}
+	}
+}
+
+func (bc *BuildController) controllerConfigWork() bool {
+	key, quit := bc.controllerConfigQueue.Get()
+	if quit {
+		return true
+	}
+	defer bc.controllerConfigQueue.Done(key)
+
+	err := bc.handleControllerConfig()
+	bc.handleControllerConfigError(err, key)
+	return false
+}
+
+// handleControllerConfig synchronizes the build controller config in the cluster
+// `build.config.openshift.io` instance and any referenced data in the `openshift-config` namespace.
+func (bc *BuildController) handleControllerConfig() error {
+	// Make sure that we have synced the controller config store
+	// This will always be false if the controller is not running on an openshift/installer cluster,
+	// and therefore does not have the service cert signing controller.
+	//
+	// TODO: Remove this once all tests run on openshift/installer clusters
+	if !waitForCacheSync(10*time.Second, bc.controllerConfigStoreSynced) {
+		glog.V(2).Info("builds.config.openshift.io not synced")
+		glog.V(4).Info("disabling service signing CA injection for builds")
+		bc.injectServiceCA = false
+		return nil
+	}
+	bc.injectServiceCA = true
+	config, err := bc.configLister.Get("cluster")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if config == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	if len(config.Spec.AdditionalTrustedCA.Name) == 0 {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	additionalCA, err := bc.configMapStore.ConfigMaps("openshift-config").Get(config.Spec.AdditionalTrustedCA.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if additionalCA == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	bc.additionalTrustedCAData = additionalCA.Data
+	return nil
+}
+
+func (bc *BuildController) handleControllerConfigError(err error, key interface{}) {
+	if err != nil {
+		glog.V(2).Infof("Failed to update controller config: %v", err)
+	}
+	bc.controllerConfigQueue.Forget(key)
+}
+
+func (bc *BuildController) controllerConfigAdded(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigUpdated(old, cur interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigDeleted(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigChanged() {
+	bc.controllerConfigQueue.Add("openshift-config")
+}
+
+func (bc *BuildController) configMapAdded(obj interface{}) {
+	configMap, ok := obj.(*corev1.ConfigMap)
```

Comment:
all these functions should be checking if event is for the configmap name we care about (whatever build config currently points to)

Reply:
@bparees updated by caching the ConfigMap name on the build controller. Better off pulling that data directly from the lister?

Reply:
caching it on the controller seems fine because the controller is going to get restarted if it ever changes, right?

Reply:
> I don't think it will restart - the controller manager's operator doesn't watch this ConfigMap.

well the configmap is moot, this is about the configmap reference field in the build config cluster resource, but you're right, i keep forgetting we redesigned this.

But basically the build controller needs to be watching the build config resource so that when things like the AdditionalCAConfigMapNameRef field is changed, the controller sees the name change.  The build controller can store that value whenever it sees it change, it doesn't need to get it from anywhere.

It's possible i'm confused about what you were originally asking.